### ISSUE TITLE
Build errors and core dump on Centos 8.3

### ISSUE THREAD

**Issue by kuiperzone** [jim-b/ECCSI-SAKKE#4]
Hi,

Unfortunately, this does not build out the box on Centos 8.3. There are a slew of warnings such as this:

src/data/communityParameters.c:187:55: warning: ‘%s’ directive output truncated writing 128 bytes into a region of size 65 [-Wformat-truncation=]
             snprintf((char *)Gx, ((tmp_glen-2)/2)+1, "%s", COMMUNITY_G+2);
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
src/data/communityParameters.c:187:13: note: ‘snprintf’ output 129 bytes into a destination of size 65
             snprintf((char *)Gx, ((tmp_glen-2)/2)+1, "%s", COMMUNITY_G+2);

Because these are treated as errors in your make, it fails. To get started, I removed the -Werror options from the make.

Now it builds, but core dumps in the demo apps with:

...
ES DEBUG: (SAKKE)                RSK:
ES DEBUG: (SAKKE)                  x:
ES DEBUG: (SAKKE)                    93AF67E5 007BA6E6 A80DA793 DA300FA4
ES DEBUG: (SAKKE)                    B52D0A74 E25E6E7B 2B3D6EE9 D18A9B5C
ES DEBUG: (SAKKE)                    5023597B D82D8062 D3401956 3BA1D25C
ES DEBUG: (SAKKE)                    0DC56B7B 979D74AA 50F29FBF 11CC2C93
ES DEBUG: (SAKKE)                    F5DFCA61 5E609279 F6175CEA DB00B58C
ES DEBUG: (SAKKE)                    6BEE1E7A 2A47C4F0 C456F052 59A6FA94
ES DEBUG: (SAKKE)                    A634A40D AE1DF593 D4FECF68 8D5FC678
ES DEBUG: (SAKKE)                    BE7EFC6D F3D68353 25B83B2C 6E69036B
ES DEBUG: (SAKKE)                  y:
ES DEBUG: (SAKKE)                    155F0A27 241094B0 4BFB0BDF AC6C670A
ES DEBUG: (SAKKE)                    65C325D3 9A069F03 659D44CA 27D3BE8D
ES DEBUG: (SAKKE)                    F311172B 55416018 1CBE94A2 A783320C
ES DEBUG: (SAKKE)                    ED590BC4 2644702C F371271E 496BF20F
ES DEBUG: (SAKKE)                    588B78A1 BC01ECBB 6559934B DD2FB65D
ES DEBUG: (SAKKE)                    2884318A 33D1A42A DF5E33CC 5800280B
ES DEBUG: (SAKKE)                    28356497 F87135BA B9612A17 26042440
ES DEBUG: (SAKKE)                    9AC15FEE 996B744C 33215123 5DECB0F5
free(): double free detected in tcache 2
Aborted (core dumped)

**Comment by jim-b** [comment]
Thanks for the info. Made some changes on a CentOS 8.4  VM using gcc 8.4.1 and openssl version 1.1.1.g

Seems the compiler is more strict now (the snprintf issues) and there have been some additional changes to openssl (re how BN_CTX end BN_CTX_free seem to work, also set_affine_coordinates_GFp has been deprecated, so changed that too).

Hope this works for you.

**Comment by kuiperzone** [comment]
Hi Jim,

Hey! That's great. We had actually figured out that BN_CTX_end() followed by BN_CTX_free() was causing a problem. We also increased the buffer size used in the logging functions which made the errors/warning go away. Moreover, as you say, we even figured out that set_affine_coordinates_GFP() was deprecated. Will try your updates.

Thanks very much. I do actually appreciate what you managed to do this with this project. And, moreover, I like the fact it has no dependencies other than openssl.

Cheers

Andy
